### PR TITLE
Add Bk2BkExpConvPV parameters for ENGINX and improve FWHM of function

### LINKED ISSUE #31691: Improve BackToBackConvPV function for EngDiff UI fitting
EngDiff UI users want to fit peaks with the BackToBackConvPV function - this requires the following

- [ ] Put A,B,S (Gamma?) from ceria runs in Parameters.xml file
- [ ] Set mixing parameter using Gaussian and Lorentzian widths automatically (possibly constrain Lorentzian and Gaussian to have same FWHM by default?)
- [ ] Improve FWHM setters in BackToBackConvPV (like done for BackToBackExponential) - this requires some assumption about the gaussian and Lorentzian widths/mixing parameter

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
**Description of work.**

- Add parameters for `Bk2BkExpConvPV` for ENGINX.
- Fix bug in pseud-voigt FWHM, `H`, and the mixing parameter `eta` - it should use the FWHM of Lorz (`2*gamma`) [1]
- Add setter for FWHM using Eq. in `BackToBackExponential` (which only has Gaussian component). Assuming the Voigt has same FWHM as Gaussian in `BackToBAckExponential` then solving Eq. for `H` assuming `H_L = H_G` (i.e. Gaussian and Lorz components of voigt have same FWHM).

[1]  P. Thompson, D. E. Cox and J. B. Hastings (1987). "Rietveld refinement of Debye-Scherrer synchrotron X-ray data from Al2O3". Journal of Applied Crystallography. 20 (2): 79–83. doi:10.1107/S0021889887087090.

**To test:**

(1) Run this script to make some data 
```
Load(Filename=r'\\isis.cclrc.ac.uk\inst$\ndxenginx\instrument\data\cycle_12_4\ENGINX00193749.nxs', OutputWorkspace='193749')
ConvertUnits(InputWorkspace='193749', OutputWorkspace='193749', Target='dSpacing')
CreateGroupingWorkspace(InputWorkspace='193749', GroupNames='NorthBank', 
    OutputWorkspace='grouping')
DiffractionFocussing(InputWorkspace='193749', OutputWorkspace='foc', 
    GroupingWorkspace='grouping')
ConvertUnits(InputWorkspace='foc', OutputWorkspace='foc', Target='TOF')

foc = ADS.retrieve('foc')
func = Bk2BkExpConvPV()
func.setParameter(0, 35219.7) # X0
func.setMatrixWorkspace(foc, 0, 0.0,0.0) # calculate A,B,S
print([func.getParamValue(ipar) for ipar in range(func.function.nParams())] )
```
It should print
`[35219.7, 0.0, 0.143624, 0.0207591, 1250.95, 9.85146]`

(2) Make this plot
```
fig, axes = plt.subplots(edgecolor='#ffffff', num='foc-1', subplot_kw={'projection': 'mantid'})
axes.plot(foc, color='#1f77b4', label='foc: spec 1', marker='.', wkspIndex=0)
axes.tick_params(axis='x', which='major', **{'gridOn': False, 'tick1On': True, 'tick2On': False, 'label1On': True, 'label2On': False, 'size': 6, 'tickdir': 'out', 'width': 1})
axes.tick_params(axis='y', which='major', **{'gridOn': False, 'tick1On': True, 'tick2On': False, 'label1On': True, 'label2On': False, 'size': 6, 'tickdir': 'out', 'width': 1})
axes.set_title('foc')
axes.set_xlabel('Time-of-flight ($\\mu s$)')
axes.set_ylabel('Counts ($\\mu s$)$^{-1}$')
axes.set_xlim([34940.0, 35654.0])
axes.set_ylim([-383.1, 45946.0])
plt.show()
```
(3) Open the fit browser and add a `Bk2BkExpConvPV` peak - note that the values of `Gamma` and `Sigma2` will be overwritten as the FWHM of the peak is estimated. Check these values `A =0.1436 `and `B=0.0207` (may not match exactly depends on the exact center of the peak)

Fixes #31691 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/CurveFitting/inc/MantidCurveFitting/Functions/Bk2BkExpConvPV.h [hunk 1]
```diff
@@ -24,7 +24,8 @@ namespace Functions {
 */
 class MANTID_CURVEFITTING_DLL Bk2BkExpConvPV : virtual public API::IPeakFunction, virtual public API::IFunctionMW {
 public:
-  Bk2BkExpConvPV();
+  /// Default constructor.
+  Bk2BkExpConvPV() : API::IPeakFunction() {}
```
Comment (StephenSmith25) [inline Framework/CurveFitting/inc/MantidCurveFitting/Functions/Bk2BkExpConvPV.h:28]: Just a minor comment, but this isn't necessary as the default constructor for each base class will be implictly called. In this case you can get away without defining a constructor, or explictly defaulting the constructor, e.g

```suggestion
  Bk2BkExpConvPV() = default;
```

**Review by StephenSmith25 — COMMENTED** [review]
Functions well! Just one comment that doesn't necessarily need attention. Also, does this need release notes?

File: Framework/CurveFitting/inc/MantidCurveFitting/Functions/Bk2BkExpConvPV.h (hunk 1 above)
Comment (RichardWaiteSTFC) [reply Framework/CurveFitting/inc/MantidCurveFitting/Functions/Bk2BkExpConvPV.h]: Thanks for spotting this, I have done as you suggested and explicitly called the default constructor. Also I have added release notes.

**Review by StephenSmith25 — DISMISSED** [review]
Looks good

File: instrument/ENGIN-X_Parameters.xml [hunk 2]
```diff
@@ -10,32 +10,21 @@ the create a distinct ENTITY for each bank-->
         <!ENTITY sigma_0_sq "0.0">
         <!ENTITY sigma_1_sq "82.1321">
         <!ENTITY sigma_2_sq "42.5988">
+		<!ENTITY PVgamma_0 "0.0">
```
Comment (gvardany) [inline instrument/ENGIN-X_Parameters.xml:13]: Could you fix the indentation here?

Comment (RichardWaiteSTFC) [reply instrument/ENGIN-X_Parameters.xml]: Thanks for spotting this, it was a tab vs spaces thing - fixed now!

**Review by StephenSmith25 — DISMISSED** [review]

File: Framework/CurveFitting/src/Functions/Bk2BkExpConvPV.cpp [hunk 3]
```diff
@@ -75,25 +72,44 @@ double Bk2BkExpConvPV::height() const {
...
 double Bk2BkExpConvPV::fwhm() const {
-  double sigma2 = this->getParameter("Sigma2");
-  double gamma = this->getParameter("Gamma");
+  // get sigma of Gauss with same FWHM as voigt (H)
   double H, eta;
-  calHandEta(sigma2, gamma, H, eta);
-
-  return mFWHM;
+  calHandEta(getParameter("Sigma2"), getParameter("Gamma"), H, eta);
+  const auto s = H / (2 * sqrt(2 * M_LN2));
```
Comment (DavidFair) [inline Framework/CurveFitting/src/Functions/Bk2BkExpConvPV.cpp:82]: Could we document in comments what were doing here, e.g. a paper / a worked example / or a commonly used operation (e.g. take the mean of x).

This tends to be "obvious" to a reader now, but later on people have to sit and mentally perform the maths each time they work on the code to understand it.

File: Framework/CurveFitting/src/Functions/Bk2BkExpConvPV.cpp [hunk 4]
```diff
@@ -75,25 +72,44 @@ double Bk2BkExpConvPV::height() const {
...
+  const auto w0 = expWidth();
+  // Gaussian and B2B exp widths don't add in quadrature. The following
+  // tends to gaussian at large S and at S=0 is equal to the intrinsic width of
+  // the B2B exp (good to <3% for typical params)
+  return w0 * exp(-0.5 * M_LN2 * s / w0) + 2 * sqrt(2 * M_LN2) * s;
```
Comment (DavidFair) [inline Framework/CurveFitting/src/Functions/Bk2BkExpConvPV.cpp:87]: ^

**Review by DavidFair — CHANGES_REQUESTED** [review]
Some code documentation changes.

Additionally, since this changes the expected results is it worth contacting known users of this that they will need to possibly re-process their data?

File: Framework/CurveFitting/src/Functions/Bk2BkExpConvPV.cpp (hunk 3 above)
Comment (RichardWaiteSTFC) [reply Framework/CurveFitting/src/Functions/Bk2BkExpConvPV.cpp]: Good point - it is actually partially explained on the comment a couple lines above, but I have commented on that line (L82) explicitly what I'm doing (converting FWHM to sigma of a Gaussian).

File: Framework/CurveFitting/src/Functions/Bk2BkExpConvPV.cpp (hunk 4 above)
Comment (RichardWaiteSTFC) [reply Framework/CurveFitting/src/Functions/Bk2BkExpConvPV.cpp:88]: I have clarified that this is the same Eq. used in` BackToBackExponential `- which I found myself, no reference required!

**Comment by RichardWaiteSTFC** [comment]
> Additionally, since this changes the expected results is it worth contacting known users of this that they will need to possibly re-process their data?

Yes they should be informed, but I don't think anyone was actually using the function, it has only very recently been working (since PR #31860) - the author of that PR is aware of these changes and I have also informed Pas.

**Review by DavidFair — APPROVED** [review]

**Pull request merged**